std/math/trig.d

```diff
@@ -0,0 +1,1281 @@
+        z = y - highBitsFactor * floor(z);
+
+        // Integer and fraction part modulo one octant.
+        int j = cast(int)(z);
+    }
+
+    // Map zeros and singularities to origin.
+    if (j & 1)
+    {
+        j += 1;
+        y += cast(T) 1.0;
+    }
+
+    z = ((x - y * P1) - y * P2) - y * P3;
+    const T zz = z * z;
+
+    enum T zzThreshold = (realFormat == RealFormat.ieeeSingle ? 1.0e-4L :
+                          realFormat == RealFormat.ieeeDouble ? 1.0e-14L : 1.0e-20L);
+    if (zz > zzThreshold)
+    {
+        static if (realFormat == RealFormat.ieeeSingle)
+            y = z + z * (zz * poly(zz, P));
+        else
+            y = z + z * (zz * poly(zz, P) / poly(zz, Q));
+    }
+    else
+        y = z;
+
+    if (j & 2)
+        y = (cast(T) -1.0) / y;
+
+    return (sign) ? -y : y;
+}
+
+@safe @nogc nothrow unittest
+{
+    static void testTan(T)()
+    {
+        // ±0
+        const T zero = 0.0;
+        assert(isIdentical(tan(zero), zero));
+        assert(isIdentical(tan(-zero), -zero));
+        // ±∞
+        const T inf = T.infinity;
+        assert(isNaN(tan(inf)));
+        assert(isNaN(tan(-inf)));
+        // NaN
+        const T specialNaN = NaN(0x0123L);
+        assert(isIdentical(tan(specialNaN), specialNaN));
+
+        static immutable T[2][] vals =
+        [
+            // angle, tan
+            [   .5,  .54630248984],
+            [   1,   1.5574077247],
+            [   1.5, 14.101419947],
+            [   2,  -2.1850398633],
+            [   2.5,-.74702229724],
+            [   3,  -.14254654307],
+            [   3.5, .37458564016],
+            [   4,   1.1578212823],
+            [   4.5, 4.6373320546],
+            [   5,  -3.3805150062],
+            [   5.5,-.99558405221],
+            [   6,  -.29100619138],
+            [   6.5, .22027720035],
+            [   10,  .64836082746],
+
+            // special angles
+            [   PI_4,   1],
+            //[   PI_2,   T.infinity], // PI_2 is not _exactly_ pi/2.
+            [   3*PI_4, -1],
+            [   PI,     0],
+            [   5*PI_4, 1],
+            //[   3*PI_2, -T.infinity],
+            [   7*PI_4, -1],
+            [   2*PI,   0],
+         ];
+
+        foreach (ref val; vals)
+        {
+            T x = val[0];
+            T r = val[1];
+            T t = tan(x);
+
+            //printf("tan(%Lg) = %Lg, should be %Lg\n", cast(real) x, cast(real) t, cast(real) r);
+            assert(isClose(r, t, CommonDefaultFor!(T,T), CommonDefaultFor!(T,T)));
+
+            x = -x;
+            r = -r;
+            t = tan(x);
+            //printf("tan(%Lg) = %Lg, should be %Lg\n", cast(real) x, cast(real) t, cast(real) r);
+            assert(isClose(r, t, CommonDefaultFor!(T,T), CommonDefaultFor!(T,T)));
+        }
+    }
+
+    import std.meta : AliasSeq;
+    foreach (T; AliasSeq!(real, double, float))
+        testTan!T();
+
+    assert(equalsDigit(tan(PI / 3), sqrt(3.0L), useDigits));
```

Comment:
CircleCI is running into problems here, the linker does not know about `equalsDigit`. It's defined in `std.math.package.d` inside of a `version (StdUnittest) package` block with visibility `package`. Should be available for the linker in my eyes. I could need help here, because I don't see, what could be wrong.

Reply:
try `package(std)` instead. That should leave the behaviour unchanged.

Reply:
Doesn't work... I think it's got to do with this `version` stuff. I guess, that `std.math` is compiled without `version(StdUnittest)` while the check for coverage in `std.math.trig` also checks the unittests. I'll try `public`. If that still produces the same error messages, we know, it's not the visibility...